internal/guest/bridge/bridge_v2.go

```diff
@@ -133,16 +136,41 @@ func (b *Bridge) createContainerV2(r *Request) (_ RequestResponse, err error) {
 //
 // This is allowed only for protocol version 4+, schema version 2.1+
 func (b *Bridge) startContainerV2(r *Request) (_ RequestResponse, err error) {
-	_, span := trace.StartSpan(r.Context, "opengcs::bridge::startContainerV2")
+	var (
+		request prot.MessageBase
+		c       *hcsv2.Container
+		cg      cgroups.Cgroup
+	)
+	ctx, span := trace.StartSpan(r.Context, "opengcs::bridge::startContainerV2")
 	defer span.End()
 	defer func() { oc.SetSpanStatus(span, err) }()
-	span.AddAttributes(trace.StringAttribute("cid", r.ContainerID))
+	span.AddAttributes(trace.StringAttribute(logfields.ContainerID, r.ContainerID))
+	entity := log.G(ctx).WithField(logfields.ContainerID, request.ContainerID)
 
 	// This is just a noop, but needs to be handled so that an error isn't
 	// returned to the HCS.
-	var request prot.MessageBase
-	if err := commonutils.UnmarshalJSONWithHresult(r.Message, &request); err != nil {
-		return nil, errors.Wrapf(err, "failed to unmarshal JSON in message \"%s\"", r.Message)
+	if err = commonutils.UnmarshalJSONWithHresult(r.Message, &request); err != nil {
+		err = errors.Wrapf(err, "failed to unmarshal JSON in message \"%s\"", r.Message)
+		return
```

Comment:
Why do we need to assign err to a variable above before returning now?

Reply:
so that the new `err` variable is used as the return and is used by the deferred `SetSpanStatus` call

Reply:
Would this not still be picked up by the deferred call if you just returned fmt.Errorf("error") directly? https://play.golang.org/p/JWMD_TQJmhP

Reply:
ohh, cause the defer happens after the return and the assignment happens
ill change this